docs/grammar/SolidityLexer.g4

```diff
@@ -0,0 +1,319 @@
+Storage: 'storage';
+String: 'string';
+Struct: 'struct';
+True: 'true';
+Try: 'try';
+Type: 'type';
+Ufixed: 'ufixed' | ('ufixed' [0-9]+ 'x' [0-9]+);
+/**
+ * Sized unsigned integer types.
+ * uint is an alias of uint256.
+ */
+UnsignedIntegerType:
+	'uint' | 'uint8' | 'uint16' | 'uint24' | 'uint32' | 'uint40' | 'uint48' | 'uint56' | 'uint64' |
+	'uint72' | 'uint80' | 'uint88' | 'uint96' | 'uint104' | 'uint112' | 'uint120' | 'uint128' |
+	'uint136' | 'uint144' | 'uint152' | 'uint160' | 'uint168' | 'uint176' | 'uint184' | 'uint192' |
+	'uint200' | 'uint208' | 'uint216' | 'uint224' | 'uint232' | 'uint240' | 'uint248' | 'uint256';
+Using: 'using';
+View: 'view';
+Virtual: 'virtual';
+While: 'while';
+
+LParen: '(';
+RParen: ')';
+LBrack: '[';
+RBrack: ']';
+LBrace: '{';
+RBrace: '}';
+Colon: ':';
+Semicolon: ';';
+Period: '.';
+Conditional: '?';
+Arrow: '=>';
+
+Assign: '=';
+AssignBitOr: '|=';
+AssignBitXor: '^=';
+AssignBitAnd: '&=';
+AssignShl: '<<=';
+AssignSar: '>>=';
+AssignShr: '>>>=';
+AssignAdd: '+=';
+AssignSub: '-=';
+AssignMul: '*=';
+AssignDiv: '/=';
+AssignMod: '%=';
+
+Comma: ',';
+Or: '||';
+And: '&&';
+BitOr: '|';
+BitXor: '^';
+BitAnd: '&';
+Shl: '<<';
+Sar: '>>';
+Shr: '>>>';
+Add: '+';
+Sub: '-';
+Mul: '*';
+Div: '/';
+Mod: '%';
+Exp: '**';
+
+Equal: '==';
+NotEqual: '!=';
+LessThan: '<';
+GreaterThan: '>';
+LessThanOrEqual: '<=';
+GreaterThanOrEqual: '>=';
+Not: '!';
+BitNot: '~';
+Inc: '++';
+Dec: '--';
+
+/**
+ * A single quoted string literal.
+ */
+StringLiteral: '"' DoubleQuotedStringCharacter* '"' | '\'' SingleQuotedStringCharacter* '\'';
+/**
+ * A single non-empty quoted string literal.
+ */
+NonEmptyStringLiteral: '"' DoubleQuotedStringCharacter+ '"' | '\'' SingleQuotedStringCharacter+ '\'';
+//@doc:inline
+fragment DoubleQuotedStringCharacter: [\u0020-\u0021\u0023-\u005B\u005D-\u007E] | UnicodeEscapeSequence | ('\\' [\u0020-\u0074\u0076-\u007E]);
+//@doc:inline
+fragment SingleQuotedStringCharacter: [\u0020-\u0026\u0028-\u005B\u005D-\u007E] | UnicodeEscapeSequence | ('\\' [\u0020-\u0074\u0076-\u007E]);
+UnicodeStringLiteral:
+    'unicode"' DoubleQuotedUnicodeStringCharacter* '"'
+    | 'unicode\'' SingleQuotedUnicodeStringCharacter* '\'';
+//@doc:inline
+fragment DoubleQuotedUnicodeStringCharacter: ~["\r\n\\] | UnicodeEscapeSequence | ('\\' ~[u]);
+//@doc:inline
+fragment SingleQuotedUnicodeStringCharacter: ~['\r\n\\] | UnicodeEscapeSequence | ('\\' ~[u]);
+/**
+ * Unicode escape sequences consist of \u followed by four hex digits.
+ */
+fragment UnicodeEscapeSequence: '\\' 'u' HexCharacter HexCharacter HexCharacter HexCharacter;
+
+/**
+ * Hex strings need to consist of an even number of hex digits that may be grouped using underscores.
+ */
+HexString: 'hex' (('"' EvenHexDigits? '"') | ('\'' EvenHexDigits? '\''));
+/**
+ * Hex numbers consist of a prefix and an arbitrary number of hex digits that may be delimited by underscores.
+ */
+HexNumber: '0' [xX] HexDigits;
```

Comment:
Why do we allow `0X11` though...

Reply:
Yeah, I also just checked - we used to but don't anymore, I'll fix it.